pandas/core/arrays/integer.py

```diff
@@ -549,21 +549,23 @@ def _reduce(self, name, skipna=True, **kwargs):
         mask = self._mask
 
         # coerce to a nan-aware float if needed
-        if mask.any():
-            data = self._data.astype("float64")
-            # We explicitly use NaN within reductions.
-            data[mask] = np.nan
+        # (we explicitly use NaN within reductions)
+        if self._hasna:
+            data = self.to_numpy("float64", na_value=np.nan)
 
         op = getattr(nanops, "nan" + name)
         result = op(data, axis=0, skipna=skipna, mask=mask, **kwargs)
 
+        if np.isnan(result):
+            return libmissing.NA
+
         # if we have a boolean op, don't coerce
         if name in ["any", "all"]:
             pass
 
         # if we have a preservable numeric op,
         # provide coercion back to an integer type if possible
-        elif name in ["sum", "min", "max", "prod"] and notna(result):
+        elif name in ["sum", "min", "max", "prod"]:
```

Comment:
Do you think this case is tested for both NA and not-NA results?

Reply:
Yes, in the `TestNumericReduce` I needed to patch below to check for pd.NA instead of np.nan, all reductions are tested with both skipna=True and False, and the "data" in those base extension tests include some missing data (at least in our tests for integer/boolean)